src/main/java/com/hedvig/contract/translators/ProductToContractEventListener.kt

```diff
@@ -138,23 +142,66 @@ class ProductToContractEventListener(
       }
 
       val contractId = fetchOrCreateContractId(e.memberId)
-
       val latestCompatAgreement = getLatestCompatAgreement(e.memberId, timeOfKnowledge)
 
       if (latestCompatAgreement != null && !latestCompatAgreement.skipAgreement) {
+        if (latestCompatAgreement.replacePreviousAgreement) {
+          try {
+            commandBus.sendAndWait<Void>(
+              ReplaceAgreement(
+                contractId = contractId,
+                oldAgreementId = latestCompatAgreement.replacingProductId!!,
+                newAgreement = ApartmentAgreement.fromCompatAgreement(latestCompatAgreement),
```

Comment:
Don't forget house! It's as easy as:

```suggestion
                newAgreement = when(latestCompatAgreement.type) {
                HOUSE -> HouseAgreement.fromCompatAgreement(latestCompatAgreement)
                else -> ApartmentAgreement.fromCompatAgreement(latestCompatAgreement)
              
```